### PR TITLE
fix: remove redundant conditon for job in release pipeline

### PR DESCRIPTION
**Pull request by tadelesh**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by seankane-msft — APPROVED** [review]
Can you verify this works by releasing `aztemplate` from this branch before merging?

File: eng/pipelines/templates/jobs/archetype-go-release.yml [hunk 1]
```diff
@@ -31,7 +31,6 @@ stages:
     jobs: 
       - deployment: TagRepository
         displayName: "Create release tag"
-        condition: and(succeeded(), eq(variables['NeedToRelease'], 'true'), ne(variables['Skip.TagRepository'], 'true'))
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-go-release.yml:34]: We only want to remove the "NeedToRelease" part of this condition.

Comment (tadelesh) [reply eng/pipelines/templates/jobs/archetype-go-release.yml:34]: `Skip.TagRepository` has already been checked with previous stage `CheckRelease` which is the dependency of this `Release` stage. So I removed it.

**Comment by tadelesh** [comment]
> Can you verify this works by releasing `aztemplate` from this branch before merging?

Release: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1224938&view=results
Skip Release: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1224955&view=results

**Comment by seankane-msft** [comment]
> > Can you verify this works by releasing `aztemplate` from this branch before merging?
> 
> Release: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1224938&view=results

The Release pipeline failed on the `Push changes` step, that definition is in `eng/common`, @weshaggard is this a flaky issue or is there anything you see that caused this failure?

File: eng/pipelines/templates/jobs/archetype-go-release.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/templates/jobs/archetype-go-release.yml:34]: Maybe I'm missing it but I don't see where `Skip.TagRepository` is checked. Can you point me to the place we check it?

**Comment by weshaggard** [comment]
That failure is because it was ran from @tadelesh's fork and test pipeline and not our official template pipeline. @tadelesh can you use our `go - aztemplate` (https://dev.azure.com/azure-sdk/internal/_build?definitionId=3058) to test? You might need to manually queue it and point it at `refs/pull/16355/merge` as the branch to test it because some things will not work in our release pipelines if triggered directly from a PR.

**Comment by tadelesh** [comment]
> That failure is because it was ran from @tadelesh's fork and test pipeline and not our official template pipeline. @tadelesh can you use our `go - aztemplate` (https://dev.azure.com/azure-sdk/internal/_build?definitionId=3058) to test? You might need to manually queue it and point it at `refs/pull/16355/merge` as the branch to test it because some things will not work in our release pipelines if triggered directly from a PR.

Thanks for your suggestion which also help me to know another way to test pipeline change under PR. Release pipeline [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1227441&view=results). Skip release pipeline [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1227450&view=results).

**Comment by seankane-msft** [comment]
> > That failure is because it was ran from @tadelesh's fork and test pipeline and not our official template pipeline. @tadelesh can you use our `go - aztemplate` (https://dev.azure.com/azure-sdk/internal/_build?definitionId=3058) to test? You might need to manually queue it and point it at `refs/pull/16355/merge` as the branch to test it because some things will not work in our release pipelines if triggered directly from a PR.
> 
> Thanks for your suggestion which also help me to know another way to test pipeline change under PR. Release result is [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1227441&view=results). Some validation failed for the `create pull request for new version` step. I don't know whether it is due to the merge ref as I haven't changed anything about the `Update package version` job. Skip release result is [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1227450&view=results). If there is no more serious problem with this PR, I want to merge it asap as it block the release of some mgmt. SDK.

@weshaggard is the failure on the Release result because it was released from a branch and not from the `main` branch? Once that issue is fixed, this looks good to me.

**Comment by weshaggard** [comment]
Yes the failure is because we cannot create a pull request against a PR merge that is expected.

@tadelesh once you handle my other feedback about the `Skip.TagRepository` this change looks good to me.

**Comment by weshaggard** [comment]
@tadelesh while in here can you also address the warning I saw in the logs https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1227441&view=logs&j=f4752989-c568-5678-80f2-0a67848bfa4d. You need to declare the machine pool and image in the CheckRelease stage similar to the other release stage otherwise it uses a default image.

File: eng/pipelines/templates/jobs/archetype-go-release.yml (hunk 1 above)
Comment (tadelesh) [reply eng/pipelines/templates/jobs/archetype-go-release.yml:34]: Sorry. I've mixed the two conditions (`Skip.Release`, `Skip.TagRepository`) up 😟. I've fixed it.

**Comment by tadelesh** [comment]
> @tadelesh while in here can you also address the warning I saw in the logs https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1227441&view=logs&j=f4752989-c568-5678-80f2-0a67848bfa4d. You need to declare the machine pool and image in the CheckRelease stage similar to the other release stage otherwise it uses a default image.

Fixed. Test log is [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1230686&view=logs&j=f4752989-c568-5678-80f2-0a67848bfa4d).

**Review by seankane-msft — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

**Pull request merged**